scripts/uniffi_bindgen_generate_kotlin_android.sh

```diff
@@ -1,17 +1,70 @@
 #!/bin/bash
+
 BINDINGS_DIR="bindings/kotlin"
 TARGET_DIR="target"
 PROJECT_DIR="ldk-node-android"
 PACKAGE_DIR="org/lightningdevkit/ldknode"
 UNIFFI_BINDGEN_BIN="cargo run --manifest-path bindings/uniffi-bindgen/Cargo.toml"
-ANDROID_NDK_ROOT="/opt/homebrew/share/android-ndk"
-LLVM_ARCH_PATH="darwin-x86_64"
+
+# The `export`s are required to make the variables available to subprocesses.
+export CFLAGS="-D__ANDROID_MIN_SDK_VERSION__=21"
+export AR=llvm-ar
+
+OS=$(uname -s)
+
+# TODO: Remove setting OPENSSL_DIR after BDK upgrade. Depends on: https://github.com/lightningdevkit/ldk-node/issues/195
+# The OPENSSL_DIR configuration is currently necessary due to default features being enabled in our esplora-client.
+# Check and set OPENSSL_DIR
+if [ -z "$OPENSSL_DIR" ]; then
+    case "$OS" in
+        Darwin)
+            # macOS
+            export OPENSSL_DIR="/opt/homebrew/Cellar/openssl@3/3.1.4"
+            ;;
+        Linux)
+            # Linux
+            export OPENSSL_DIR="/usr/local/openssl"
+            ;;
+        *)
+    esac
+fi
+
+# Check and set ANDROID_NDK_ROOT
+if [ -z "$ANDROID_NDK_ROOT" ]; then
```

Comment:
Added logic to set the other variables depending on OS, but I'm not really happy with the current solution - don't think it's robust nor scalable. Any ideas?